src/engine/source/api/src/metrics/handlers.cpp

```diff
@@ -0,0 +1,210 @@
+#include "api/metrics/handlers.hpp"
+
+#include <json/json.hpp>
+#include <eMessages/eMessage.h>
+#include <eMessages/metrics.pb.h>
+
+#include <api/adapter.hpp>
+
+namespace api::metrics::handlers
+{
+
+namespace eMetrics = ::com::wazuh::api::engine::metrics;
+namespace eEngine = ::com::wazuh::api::engine;
+
+/* Manager Endpoint */
+
+api::Handler metricsDumpCmd(const std::shared_ptr<metricsManager::IMetricsManagerAPI>& metricsAPI)
+{
+    return [metricsAPI](api::wpRequest wRequest) -> api::wpResponse
+    {
+        using RequestType = eMetrics::Dump_Request;
+        using ResponseType = eMetrics::Dump_Response;
+        auto res = ::api::adapter::fromWazuhRequest<RequestType, ResponseType>(wRequest);
+
+        // If the request is not valid, return the error
+        if (std::holds_alternative<api::wpResponse>(res))
+        {
+            return std::move(std::get<api::wpResponse>(res));
+        }
+
+        // Validate the params request
+        const auto& eRequest = std::get<RequestType>(res);
+        auto result = metricsAPI->dumpCmd();
+
+        if (std::holds_alternative<base::Error>(result))
+        {
+            return ::api::adapter::genericError<ResponseType>(std::get<base::Error>(result).message);
+        }
+
+        const auto protoVal = eMessage::eMessageFromJson<google::protobuf::Value>(std::get<std::string>(result));
+        const auto json_value = std::get<google::protobuf::Value>(protoVal);
+
+        ResponseType eResponse;
+        eResponse.set_status(eEngine::ReturnStatus::OK);
+        eResponse.mutable_value()->CopyFrom(json_value);
+
+        return ::api::adapter::toWazuhResponse(eResponse);
+    };
+}
+
+
+api::Handler metricsGetCmd(const std::shared_ptr<metricsManager::IMetricsManagerAPI>& metricsAPI)
```

Comment:
When getting a metric the name of the instrument appears in the scope field, is this correct?
```
$ sudo ./build/main metrics get Router UsedQueue
{"scope":"UsedQueue","version":"","records":[{"attributes":[{"type":"LastValuePointData","valid":true,"timestamp":"1681117936122842915","value":0}],"type":"ObservableGauge","instrument_description":"","start_time":"Mon Apr 10 09:12:16 2023","unit":"","instrument_name":"UsedQueue"}],"schema":""}
```